src/test/v1.2/advanced/channels-support-1.2.ts

```diff
@@ -105,13 +105,17 @@ export class ChannelControl1_2 implements ChannelControl<Channel, Context> {
 
   setupAndValidateListener1 = (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
     if (channel) {
-      listener1 = channel.addContextListener(null, (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener1 = channel.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     } else {
-      listener1 = fdc3.addContextListener(null, (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener1 = fdc3.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     }
```

Comment:
This isn't quite right as we're using the same argument for both the listener type and the expected type. We need to be able to listen for null but expect a particular type in many tests...

Due to the deleted lines I can't put a suggestion in here but I think this function should be:

```typescript
setupAndValidateListener1 = (channel: Channel, listenerContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
    if (channel) {
      listener1 = channel.addContextListener(listenerContextType, (context) => {
        expect(context.type).to.be.equals(expectedContextType, errorMessage);
        onComplete(context);
      });
    } else {
      listener1 = fdc3.addContextListener(listenerContextType, (context) => {
        expect(context.type).to.be.equals(expectedContextType, errorMessage);
        onComplete(context);
      });
    }
    validateListenerObject(listener1);
  }
  ```

same applies to the second copy of the function below